### PR TITLE
nvidia-prime: add package

### PR DESCRIPTION
**Pull request by OriginCode**
The script still doesn't work.

Ref: https://t.me/c/1096355081/386199 https://t.me/c/1096355081/384364

### CODE REVIEW

File: extra-x11/nvidia-prime/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,5 @@
+abinfo "Fetching prime-run binary..."
+wget -O prime-run https://git.archlinux.org/svntogit/packages.git/plain/trunk/prime-run\?h\=packages/nvidia-prime
```
Comment (MingcongBai) [inline extra-x11/nvidia-prime/autobuild/build]: Use absolute path here, `wget -O "$SRCDIR"/prime-run ...`.

File: extra-x11/nvidia-prime/autobuild/build [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+abinfo "Installing prime-run binary..."
+install -Dvm755 prime-run "$PKGDIR"/usr/bin/prime-run
```
Comment (MingcongBai) [inline extra-x11/nvidia-prime/autobuild/build]: Same as above.

File: extra-x11/nvidia-prime/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=nvidia-prime
+PKGSEC=x11
+PKGDES="NVIDIA Prime Render Offload configuration and utilities"
+PKGDEP="libglvnd mesa vulkan"
```
Comment (MingcongBai) [inline extra-x11/nvidia-prime/autobuild/defines]: Is `nvidia-prime` a "binary" or a script? If it's a script, append `ABHOST=noarch` as below (same goes for all packages marked as "any" in Arch Linux.

File: extra-x11/nvidia-prime/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=nvidia-prime
+PKGSEC=x11
+PKGDES="NVIDIA Prime Render Offload configuration and utilities"
```
Comment (MingcongBai) [inline extra-x11/nvidia-prime/autobuild/defines]: Considerations:

- Is "NVIDIA Prime Render Offload" a proper noun? AFAIK, only NVIDIA and "Prime" are, if I'm not mistaken, please do not capitalise the others.
- Is it "Prime," or "PRIME"?

File: extra-x11/nvidia-prime/autobuild/overrides/usr/share/X11/xorg.conf.d/20-nvidia-prime.conf [hunk 5]
```diff
@@ -0,0 +1,4 @@
+Section "ServerLayout"
```
Comment (MingcongBai) [inline extra-x11/nvidia-prime/autobuild/overrides/usr/share/X11/xorg.conf.d/20-nvidia-prime.conf:6]: Please add some comments describing this configuration's functionalities, and note that it comes from a package (`nvidia-prime`) in this case.

File: extra-x11/nvidia-prime/autobuild/overrides/usr/share/X11/xorg.conf.d/20-nvidia-prime.conf [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+    Identifier "layout"
+    Option "AllowNVIDIAGPUScreens"
+EndSection
```
Comment (MingcongBai) [inline extra-x11/nvidia-prime/autobuild/overrides/usr/share/X11/xorg.conf.d/20-nvidia-prime.conf:9]: Should this file be modified by users? If so, add `/usr/share/X11/xorg.conf.d/20-nvidia-prime.conf` to `autobuild/conffiles`.

File: extra-x11/nvidia-prime/spec [hunk 7]
```diff
@@ -0,0 +1,2 @@
+VER=1.0
+DUMMYSRC=1
```
Comment (MingcongBai) [inline extra-x11/nvidia-prime/spec:2]: Don't use `DUMMYSRC=1` unless you absolutely have to, and in this case, you don't.

File: extra-x11/nvidia-prime/autobuild/defines (hunk 3 above)
Comment (MingcongBai) [inline extra-x11/nvidia-prime/autobuild/defines]: I thought you mentioned that it depends on `nvidia-utils` or something?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-x11/nvidia-prime/autobuild/defines (hunk 3 above)
Comment (OriginCode) [reply extra-x11/nvidia-prime/autobuild/defines]: Yes. It depends on `nvidia-utils` which provides `nvidia-libgl, opengl-driver, vulkan-driver` in Arch Linux.

Comment (MingcongBai) [reply extra-x11/nvidia-prime/autobuild/defines]: Okay, in this case, please add `nvidia` to `$PKGDEP`.

File: extra-x11/nvidia-prime/autobuild/defines (hunk 4 above)
Comment (OriginCode) [reply extra-x11/nvidia-prime/autobuild/defines]: - Modified to "NVIDIA PRIME Render Offloading configuration and utilities".
- "PRIME" according to Arch Wiki.
This description was originally from the package https://www.archlinux.org/packages/extra/any/nvidia-prime/

File: extra-x11/nvidia-prime/autobuild/build (hunk 1 above)
Comment (OriginCode) [reply extra-x11/nvidia-prime/autobuild/build]: Recv

File: extra-x11/nvidia-prime/autobuild/build (hunk 2 above)
Comment (OriginCode) [reply extra-x11/nvidia-prime/autobuild/build]: Recv

File: extra-x11/nvidia-prime/autobuild/defines (hunk 3 above)
Comment (OriginCode) [reply extra-x11/nvidia-prime/autobuild/defines]: It is a script. Appended `ABHOST=noarch`.

File: extra-x11/nvidia-prime/autobuild/overrides/usr/share/X11/xorg.conf.d/20-nvidia-prime.conf (hunk 6 above)
Comment (OriginCode) [reply extra-x11/nvidia-prime/autobuild/overrides/usr/share/X11/xorg.conf.d/20-nvidia-prime.conf:9]: I think this file should not be modified by users.

File: extra-x11/nvidia-prime/autobuild/defines (hunk 3 above)
Comment (OriginCode) [reply extra-x11/nvidia-prime/autobuild/defines]: 6521ddd

**Comment by MingcongBai** [comment]
Спасибо за ваш Pull Request!

**Pull request merged**